racingcar-java/src/main/java/step2/StringCalculator.java

```diff
@@ -0,0 +1,60 @@
+package step2;
+
+public final class StringCalculator extends ArithmeticCalculator {
+
+    private final static String[] operators = {"+", "-", "*", "/"};
+
+    public int calculation(final String str) {
+        String[] input =  this.validation(str);
+        String operator = null;
+        int result = Integer.parseInt(input[0]);
+
+        for (int i = 1; i < input.length; i++) {
+            if (isOperator(input[i])) {
+                operator = input[i];
+            } else {
+                int y = Integer.parseInt(input[i]);
+                result = selectArithmeticOperation(operator, result, y);
+            }
+        }
+
+        return result;
+    }
+
+    private String[] validation(final String str) throws IllegalArgumentException{
+        if (str == null || str == "") throw new IllegalArgumentException();
```

Comment:
```suggestion
    private String[] validation(final String str) throws IllegalArgumentException{
        if (str == null || str == "") {
            throw new IllegalArgumentException();
        }
```
validation 👍 적절한 예외에 대한 메시지를 남겨주는것이 어떨까요?

그리고 if문에는 중괄호를 담아주는 것이 기본 컨벤션입니다:)

Reply:
진영님 의견에 동의합니다.  
우선, 커스텀한 익셉션을 사용하는 것을 추천하는 것은 '절대' 아님을 명시해주세요  
예외에 대한 메시지가 없을 경우 디버깅하기 매우 힘들어집니다.   
더불어 위와 같이 제네럴한 익셉션을 사용했을 경우 더더욱 힘들어집니다.

Reply:
아 이부분은 제가 버릇처럼 사용하던 부분인데 고쳐야겠군요.. 좋은 지적 정말 감사합니다 👍